pinot-core/src/main/java/org/apache/pinot/core/data/table/IndexedTable.java

```diff
@@ -32,38 +32,44 @@
  * Base abstract implementation of Table for indexed lookup
  */
 public abstract class IndexedTable implements Table {
-  // When table reaches max capacity, we will allow 20% more records to get inserted (bufferedCapacity)
-  // If records beyond bufferedCapacity are received, the table will undergo sort and evict upto evictCapacity (10% more than capacity)
-  // This is to ensure that for a small number beyond capacity, a fair chance is given to all records which have the potential to climb up the order
-  /** Factor used to add buffer to maxCapacity of the Collection used **/
-  private static final double BUFFER_FACTOR = 1.2;
-  /** Factor used to decide eviction threshold **/
-  private static final double EVICTION_FACTOR = 1.1;
 
   List<AggregationFunction> _aggregationFunctions;
   DataSchema _dataSchema;
-  List<AggregationInfo> _aggregationInfos;
-  List<SelectionSort> _orderBy;
+  boolean _sort;
 
   int _maxCapacity;
-  int _evictCapacity;
   int _bufferedCapacity;
 
   @Override
   public void init(@Nonnull DataSchema dataSchema, List<AggregationInfo> aggregationInfos, List<SelectionSort> orderBy,
-      int maxCapacity) {
+      int maxCapacity, boolean sort) {
```

Comment:
Sort tells us whether the caller is expecting the results sorted. For example, at server level, no need to have results sorted, but at broker level, we need them sorted.
Sort is better off in init, because it is currently needed in iterator() or in finish(), and maybe even elsewhere

Reply:
I think you only need it in `finish()` right? Before `finish()` getting called, you never need sorted results

Reply:
Refactored to pass sort flag in finish() instead of init()